### PR TITLE
Fixes a RuntimeException when resolving a methodcall with a generic array as parameter.

### PR DESCRIPTION
**Pull request by maartenc**
Resolving the following construct will result in a RuntimeException:

```
List<String> foo = new ArrayList<>();
foo.toArray(new String[1]);

```
This PR fixes this.

### CODE REVIEW

File: javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/MethodCallExprContext.java [hunk 1]
```diff
@@ -255,6 +255,12 @@ private void inferTypes(ResolvedType source, ResolvedType target, Map<ResolvedTy
             mappings.put(target.asTypeParameter(), source);
             return;
         }
+        if (source.isArray() && target.isArray()) {
+            ResolvedType sourceComponentType = source.asArrayType().getComponentType();
+            ResolvedType targetComponentType = target.asArrayType().getComponentType();
```
Comment (matozoid) [inline javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/MethodCallExprContext.java:260]: The component type for `int[][][]` is `int[][]`, and its element type is `int`. Does this code work for multidimensional arrays too?

Comment (maartenc) [reply javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/MethodCallExprContext.java:260]: Good question.
If you change the method parameter in the test to String[][][] it seems to work...

Comment (matozoid) [reply javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/MethodCallExprContext.java:260]: Good enough for me :-)

**Comment by matozoid** [comment]
Thanks! If @ftomassetti is also happy we'll merge it.

**Comment by ftomassetti** [comment]
Infer types is such a cool feature! And it looks good to me. I just updated the branch, let's see what happens to test but for me it is good to go

**Comment by ftomassetti** [comment]
Thank you @maartenc for your awesome contributions!

**Review by ftomassetti — APPROVED** [review]
Approved, pending tests

**Pull request merged**